typedmodels/models.py

```diff
@@ -410,21 +410,21 @@ def recast(self, typ=None):
                     # Instead, we raise an error when the object is saved.
                     return
             else:
-                self.type = self._typedmodels_type
+                self.dtm_type = self._typedmodels_type
 
         if typ is None:
-            typ = self.type
+            typ = self.dtm_type
         else:
             if isinstance(typ, type) and issubclass(typ, base):
                 model_name = typ._meta.model_name
-                typ = "%s.%s" % (typ._meta.app_label, model_name)
+                typ = ContentType.objects.get(app_label=typ._meta.app_label, model=model_name)
```

Comment:
This line is a little iffy, it can probably be done without querying the DB

Reply:
https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/#django.contrib.contenttypes.models.ContentTypeManager.get_for_model

```suggestion
                typ = ContentType.objects.get_for_model(typ)
```